In [1]:
import os

import tensorflow as tf
import multiprocessing
import numpy as np
import matplotlib.pyplot as plt
from zipfile import ZipFile
import time
from IPython import get_ipython




In [43]:
from multiprocessing import Process, Manager
def inferMnist(dict):
    dict['loads'] += 1

def main():
    manager = Manager()

    img_fns = manager.dict()
    img_fns['loads'] = 0
    p1 = Process(target=inferMnist, args=(img_fns,))
    p2 = Process(target=inferMnist, args=(img_fns,))
    p1.start()
    p2.start()
    p1.join()
    p2.join()
    print(img_fns)

In [44]:
main()

{'loads': 0}


awesome


In [3]:
import Multiprocessing.multiprocess_script as mp_script
import importlib
importlib.reload(mp_script)
mp_script.main(dontRead=False)


198
time elapsed: 183.27778840065002


180 seconds for two instances inferring on 198 images.
with skips.

In [53]:
start_time = time.time()
mp_script.inferMnist({'skips':0})
print(time.time() - start_time)

167.65413403511047


In [2]:
import Multiprocessing.multiprocess_script as mp_script
import importlib
importlib.reload(mp_script)
mp_script.main(dontRead=True)

0
time elapsed: 258.17185640335083


258 sec for two instances inferring on 198 (x 2 ) images.

In [6]:
policy = tf.keras.mixed_precision.experimental.Policy('mixed_float16')
tf.keras.mixed_precision.experimental.set_policy(policy)



In [7]:
start_time = time.time()
mp_script.inferMnist({'skips':0})
print(time.time() - start_time)


168.48670148849487


Using multiprocessing correctly can boost performance by a minimum of 150% or more based on how many GPUs are used.